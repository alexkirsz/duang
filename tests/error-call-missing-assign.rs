// this test check the error message when mandatory parameter is lost.
use duang::duang;

duang!(
pub fn foo<T>(a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
where
  T: std::ops::Mul<T, Output = T>,
  T: std::fmt::Display,
  T: Copy,
{
  (a,b,c)
}
);

fn main() {
  foo!(b=0.1);
}
