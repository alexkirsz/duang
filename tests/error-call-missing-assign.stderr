error: Mandatory argument is not assigned.
  --> $DIR/error-call-missing-assign.rs:5:15
   |
5  | pub fn foo<T>(a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
   |               ^
...
16 |   foo!(b=0.1);
   |   ------------ in this macro invocation
